### PR TITLE
Timetable: add calendar filter on route_schedules API

### PR DESCRIPTION
**Pull request by eturck**

### CODE REVIEW

**Comment by buildbotctp** [comment]
Can one of the admins verify this patch?

**Comment by buildbotctp** [comment]
Can one of the admins verify this patch?

**Comment by antoine-de** [comment]
ok to test

**Comment by antoine-de** [comment]
Nice feature, thanks @eturck !

I think it's just lacking tests.
Can you add some ? you can look at the tests done for stop_schedule:
https://github.com/CanalTP/navitia/blob/7e858f75a68af847c12c3d222faaaae0c488f229/source/time_tables/tests/departure_boards_test.cpp#L189-208

You can use (copy or adapt as you wish) the fixture used: 
https://github.com/CanalTP/navitia/blob/dev/source/time_tables/tests/departure_board_test_data.h

No need to add lot's of them, but it's very useful.

I'm not sure it's very clear, if you don't feel comfortable with them, do not hesitate to ask questions (here or on IRC freenode/#navitia), or if you prefer, we can add them to your PR with another PR.

File: source/time_tables/route_schedules.cpp [hunk 1]
```diff
@@ -59,10 +60,19 @@ get_all_stop_times(const vector_idx& journey_patterns,
...
-                                      max_stop_date_times, d, disruption_active);
+    std::vector<datetime_stop_time> first_dt_st;
+    if(!calendar_id) {
+        //On fait un best_stop_time sur ces journey_pattern points
+        first_dt_st = get_stop_times(first_journey_pattern_points,
+                                          dateTime, max_datetime,
+                                          max_stop_date_times, d, disruption_active);
+    }
+    else {
+        //On fait un best_stop_time sur ces journey_pattern points
```
Comment (antoine-de) [inline source/time_tables/route_schedules.cpp]: the comment is not up to date

File: source/time_tables/tests/route_schedules_test.cpp [hunk 2]
```diff
@@ -88,7 +88,7 @@ struct route_schedule_fixture {
 
 BOOST_FIXTURE_TEST_CASE(test1, route_schedule_fixture) {
 
-    pbnavitia::Response resp = navitia::timetables::route_schedule("line.uri=A", {}, d("20120615T070000"), 86400, 100,
+    pbnavitia::Response resp = navitia::timetables::route_schedule("line.uri=A", boost::optional<const std::string>(), {}, d("20120615T070000"), 86400, 100,
```
Comment (antoine-de) [inline source/time_tables/tests/route_schedules_test.cpp]: it's not important at all but you can use `boost::none` (or `{}`)

**Comment by eturck** [comment]
Thanks for the great feedback! 

I tried to fix the comment issue and add a little bit of testing, I'm not sure if it's enough and if it's done in the proper way. If it's not I will modify it again.
The merge build seems to have fail, I'm fairly new to new pull request system and I'm not sure how to fix this and even how to have a little bit more details on what the problem is.

Let me know what I should do or if I've done something wrong =).

**Comment by eturck** [comment]
Well. It's not mark as failed anymore, and I didn't do anything :). Just let me know if I can improve anything.

**Comment by antoine-de** [comment]
Great, nice test!

Don't worry about the 'fail' mark, we are migrating our integration server, there are some problems :smile:

**Pull request merged**